builder/pipeline.cc

```diff
@@ -456,9 +456,19 @@ stmt build_pipeline(node_context& ctx, const std::vector<buffer_expr_ptr>& input
   }
   result = infer_bounds(result, ctx, input_syms);
 
-  result = fix_buffer_races(result);
+  result = simplify(result);
```

Comment:
I'm a little confused about this code motion; it seems reasonable but it's not clear to me how it's related to the visualization changes

Reply:
It enables implementing the `no_alias_buffers` option without plumbing it through layers of code. It's a good change anyways, `infer_bounds` should not be a dumping ground for lowering passes.